bug fixes, and code reorganization

MuellerMatthew opened the pull request:
-OnRemove now removes listeners the layer is removed from the map.
-Multiple options can now be passed in to the initialization function, or to the factory function.
-Options are now set using the standardized L.setOptions()
-A new option called DisplaySort is called which can be used to sort which markers should be shown on the map, or to identify which marker properties should be used when displaying data.

MuellerMatthew committed:
bug fixes, and code reorganization

-OnRemove now removes listeners the layer is removed from the map.
-Multiple options can now be passed in to the initialization function, or to the factory function.
-Options are now set using the standardized L.setOptions()
-A new option called DisplaySort is called which can be used to sort which markers should be shown on the map, or to identify which marker properties should be used when displaying data.

Eclipse1979 commented:
Hi,
First of all, thank you for your contribution.
However there seems to be some issues with the code : 
- my browser tells me there is a syntax error on line 8 (SyntaxError: missing } after property list), I guess you forgot a comma on the end of line 7 ?
- even after correcting this, on the demo I get an error with the json file which I did not have before (I'm not sure whether this is linked to your code).

Also I do not quite understand how the sorting is done. In the demo, when zooming out, the bigger circles tend to be replaced with smaller ones.

MuellerMatthew committed:
fixing typo

MuellerMatthew committed:
Update leaflet-conditionalLayer2.html

MuellerMatthew committed:
Update leaflet.conditionalLayer.js

MuellerMatthew commented:
I fixed the missing comma, and updated the default sort function so it sorts by `leaflet_id`. 

When a user wants to sort their data before selecting which markers should be shown they can pass a custom sorting function.  
For example if I wanted to sort the bubbles in the second example by radius i could pass:
`function(a, b) { return a._radius - b._radius; }` as a function to sort so only the smallest circles are shown, or i could pass: `function(a, b) { return b._radius - a._radius; }` so that only the largest circles that fall within the view are shown. This allows a user to sort based upon any values stored in the each layer. 

Regarding the error with a json file, I am not able to replicate it, and none of my changes should have altered any json files. Can you tell me which browser you were testing with?

MuellerMatthew committed:
Update README.md

MuellerMatthew committed:
Update README.md

Eclipse1979 commented:
Now everything works fine for me, except for the fact that I still get a syntax error in the console for the json on line 1 (but it doesn't seem to affect anything). I'm using Firefox 45.0.2
However I doubt that this is linked with your code.

I will merge this
Thank you again
